test/IdTableTest.cpp

```diff
@@ -985,21 +985,22 @@ TEST(IdTable, shrinkToFit) {
   // necessary to change them if one of our used standard libraries has a
   // different behavior, but this is unlikely due to ABI stability goals between
   // library versions.
-  auto memory = ad_utility::makeAllocationMemoryLeftThreadsafeObject(1000);
+  auto memory = ad_utility::makeAllocationMemoryLeftThreadsafeObject(1_kB);
   IdTable table{2, ad_utility::AllocatorWithLimit<Id>{memory}};
-  ASSERT_EQ(memory.ptr().get()->wlock()->numFreeBytes(), 1000);
+  using namespace ad_utility::memory_literals;
+  ASSERT_EQ(memory.ptr().get()->wlock()->amountMemoryLeft(), 1_kB);
```

Comment:
Can you implement this call chain as a member function, s.t. you can just write
`memory.amountMemoryLeft()`
(also propagate this member function to the `AllocatorWithLimit`.

Reply:
then we can maybe even make the `ptr()` function private, because `.get()->wlock()->` doesn't look like something that should be called from the outside.

Reply:
Alright, I took a closer look at it and I don't think, that we should change anything here.

`makeAllocationMemoryLeftThreadsafeObject` generates a `AllocationMemoryLeftThreadsafe`, which is just a very basic wrapper around a `std::shared_ptr`. The wrapper itself is basically just a `std::shared_ptr` and I honestly can't tell, why it even exists. A `using ... = ....` would deliver the same level of abstraction and functionality, while also reducing the amount of code needed for its usage.

Adding a `amountMemoryLeft()` to a basic pointer doesn't seem the best idea to me.

For the same reason, making `ptr()` private would make the entire class useless. Like, straight up. Without it, you can only compare instances for equality.

> (also propagate this member function to the `AllocatorWithLimit`.

This function already exists.

Reply:
After a short conversation, me and @joka921 , decided to scrap this issue. This whole thing would require a rather big rewrite of multiple things, that do not really have anything to do with this PR.